### PR TITLE
http server header file

### PR DESCRIPTION
**Pull request by AriaZhou**

### CODE REVIEW

File: src/Server.h [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+#include <iostream>
+#include <unordered_map>
+#include <string>
+
+#include "HttpRequestHandler.h"
+
+#define	Asio boost::asio
+#define TCP boost::asio::ip::tcp
+
+typedef boost::shared_ptr<HttpRequestHandler> handler_ptr;
```
Comment (john0312) [inline src/Server.h]: Please do not use typedef at global scope. You can put it in the class.

File: src/Server.h [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+class Server {
+public:
+
+    /**
+     * @param io_service
+     * @param ed
+     */
+    Server(Asio::io_service& io_service, TCP::endpoint& ep)
+        :io_service(io_service), acceptor(io_service, ep), sessionCount(0)
+    {
```
Comment (john0312) [inline src/Server.h]: Please move implementation to the .cc file.

**Review by john0312 — COMMENTED** [review]
Same comments as the other PRs.
Please clang format your PR.

File: src/Server.h (hunk 2 above)
Comment (AriaZhou) [reply src/Server.h]: done

File: src/Server.h (hunk 1 above)
Comment (AriaZhou) [reply src/Server.h]: done

File: src/Server.h [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+ * Project MyHttpServer
+ * @author Pei-Ru Wang, Cheryl Huang, Yuan Zhou
+ * @version 0.0.1
+ */
+
+#ifndef _SERVER_H
+#define _SERVER_H
+
+#include <iostream>
+#include <string>
```
Comment (kungyao) [inline src/Server.h:15]: Please remove unused header.

Comment (kungyao) [inline src/Server.h:14]: Please remove unused header.

File: src/Server.h [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+#include "HttpRequestHandler.h"
+
+// Responsible for receiving requests from customers and handing them to
+// requesthandler for processing. Manage customer requests.
+class Server {
+
+public:
+  // Initialize the server, establish an endpoint and wait for the connection
+  Server(boost::asio::io_service &io_service,
+         boost::asio::ip::tcp::endpoint &ep);
```
Comment (kungyao) [inline src/Server.h]: Please use const reference or pointer.

**Review by kungyao — COMMENTED** [review]
Please squash the commits for this PR.

File: src/Server.h [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+  void StartServer();
+
+private:
+  // Handle the requests sent by the client
+  void HandleAccept(handler_ptr new_handler,
+                    const boost::system::error_code &error);
+
+private:
+  // Interact with the I/O services of the OS
+  boost::asio::io_service &io_service;
```
Comment (john0312) [inline src/Server.h]: Please use pointer instead.

File: src/Server.h [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+  // Listen for connection requests
+  boost::asio::ip::tcp::acceptor acceptor;
+
+  // The number of sessions currently connected to the server
+  int sessionCount;
```
Comment (john0312) [inline src/Server.h]: Mixed naming style.

File: src/Server.h [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+
+#include "HttpRequestHandler.h"
+
+// Responsible for receiving requests from customers and handing them to
+// requesthandler for processing. Manage customer requests.
+class Server {
+
+public:
+  // Initialize the server, establish an endpoint and wait for the connection
+  Server(const std::string &address, const std::string &port,
```
Comment (john0312) [inline src/Server.h]: `uint16_t` for port.

**Review by john0312 — APPROVED** [review]

**Pull request merged**